data_steward/constants/validation/participants/validate.py

```diff
@@ -229,62 +107,145 @@
-    {{city_with_clause}}
+    WITH 
+      normalized_rdr AS ({{rdr_city}}),
+      normalized_ehr AS ({{ehr_city}})    
     SELECT
       CASE
-        WHEN normalized_rdr_city.rdr_city = normalized_ehr_city.ehr_city THEN '{{match}}'
-        WHEN normalized_rdr_city.rdr_city IS NOT NULL AND normalized_ehr_city.ehr_city IS NOT NULL THEN '{{no_match}}'
-        WHEN normalized_rdr_city.rdr_city IS NULL THEN '{{missing_rdr}}'
-        ELSE '{{missing_ehr}}'
+        WHEN rdr_city IS NULL THEN '{{missing_rdr}}'
+        WHEN ehr_city IS NULL THEN '{{missing_ehr}}'
+        WHEN normalized_rdr.city = normalized_ehr.city THEN '{{match}}'
+        ELSE '{{no_match}}'
       END AS city
-    FROM normalized_rdr_city, normalized_ehr_city));
+    FROM normalized_rdr, normalized_ehr));
+""")
+
+NORMALIZED_STREET = JINJA_ENV.from_string("""
+    WITH address_abbreviations AS (
+        SELECT *
+        FROM UNNEST(ARRAY<STRUCT<abbreviation STRING, expansion STRING>>[
+            {{abbreviation_tuples}}
+        ])
+    ),
+    remove_commas_and_periods AS (
+        SELECT 'dummy_id' as id, REGEXP_REPLACE(coalesce({{street}}, ''), '[,.]', '') as address
+    ),
+    remove_redundant_whitespaces AS (
+        SELECT id, REGEXP_REPLACE(TRIM(address), ' +', ' ') as address
+        FROM remove_commas_and_periods
+    ),
+    lowercase AS (
+        SELECT id, LOWER(address) as address
+        FROM remove_redundant_whitespaces
+    ),
+    standardize_ordinal_number AS (
+        SELECT id, REGEXP_REPLACE(address,'([0-9])(?:st|nd|rd|th)', r'\\1') as address
+        FROM lowercase
+    ),
+    standardize_apartment_number AS (
+        SELECT id, REGEXP_REPLACE(address,'([0-9])([a-z])',r'\\1 \\2') as address
+        FROM standardize_ordinal_number
+    ),
+    split_address_into_parts AS (
+        SELECT id, part_address
+        FROM standardize_apartment_number, UNNEST(SPLIT(address, ' ')) as part_address
+    ),
+    unabbreviate AS (
+        SELECT 
+            id, COALESCE(expansion, part_address) as normalized_part_address,
+        FROM split_address_into_parts p
+        LEFT JOIN address_abbreviations aa
+        ON aa.abbreviation = p.part_address
+    )
+    SELECT 
+        ARRAY_TO_STRING(ARRAY_AGG(normalized_part_address), ' ') as street,
+    FROM unabbreviate
+    GROUP BY id
+""")
+
+NORMALIZED_CITY = JINJA_ENV.from_string("""
+    WITH address_abbreviations AS (
+        SELECT *
+        FROM UNNEST(ARRAY<STRUCT<abbreviation STRING, expansion STRING>>[
+            {{abbreviation_tuples}}
+        ])
+    ),
+    remove_commas_and_periods AS (
+        SELECT 'dummy_id' as id, REGEXP_REPLACE(coalesce({{city}}, ''), '[,.]', '') as address
+    ),
+    remove_redundant_whitespaces AS (
+        SELECT id, REGEXP_REPLACE(TRIM(address), ' +', ' ') as address
+        FROM remove_commas_and_periods
+    ),
+    lowercase AS (
+        SELECT id, LOWER(address) as address
+        FROM remove_redundant_whitespaces
+    ),
```

Comment:
This may be too modular, you can add this to any of the above. Also, all the string manipulation could potentially be in one, and directly placed in the CREATE_CITY_COMPARISON_FUNCTION Jinja template. Similar for street. However, let me know if that's not as doable.